src/coreclr/jit/morph.cpp

```diff
@@ -14460,25 +14417,26 @@ void Compiler::fgExpandQmarkNodes()
         {
             for (Statement* const stmt : block->Statements())
             {
-                GenTree* expr = stmt->GetRootNode();
-#ifdef DEBUG
-                fgPreExpandQmarkChecks(expr);
-#endif
+                INDEBUG(fgPreExpandQmarkChecks(stmt->GetRootNode()));
                 introducedThrows |= fgExpandQmarkStmt(block, stmt);
             }
         }
-#ifdef DEBUG
-        fgPostExpandQmarkChecks();
-#endif
+
+        fgInvalidateDfsTree();
+        m_dfsTree = fgComputeDfs();
     }
+
     compQmarkRationalized = true;
+    INDEBUG(fgPostExpandQmarkChecks());
 
     // TODO: if qmark expansion created throw blocks, try and merge them
     //
     if (introducedThrows)
```

Comment:
I guess `introducedThrows` can be removed, doesn't seem to add value (or can be replaced with `bool modified = `